core/src/App.svelte

```diff
@@ -924,6 +929,10 @@
    * @param goBackContext the goBack context that is passed through when closing the modal
    */
   const closeModal = (index, isClosedInternal, goBackContext) => {
+    if (mfModalList?.length && !mfModalList[index]) {
+      index = mfModalList.length - 1;
+    }
```

Comment:
the **closeModal** function here, must '_close the modal given the respective modal index._' as described in the description. The given index should be given correctly. If this method receives an incorrect index, the function should fail. This lets us know, that we call this function with incorrect parameters somewhere. I would remove this check here and find the issue someplace else, where needed.